pymc4/ast_compiler.py

```diff
@@ -114,26 +103,68 @@ def parse_snippet(source, filename, mode, flags, firstlineno, privateprefix_igno
     return a
 
 
-class AutoNameTransformer(ast.NodeTransformer):
-    def visit_Assign(self, tree_node):
-        try:
-            rv_name = tree_node.targets[0].id
-            # Test if creation of known RV
-            func = tree_node.value.func
-            if hasattr(func, "attr"):
-                call = func.attr
-            else:
-                call = func.id
-
-            if call not in ALL_RVs:
-                return tree_node
-
-            # Test if name keyword is already set
-            if any(kwarg.arg == "name" for kwarg in tree_node.value.keywords):
-                return tree_node
-            else:
-                tree_node.value.keywords.append(ast.keyword("name", ast.Str(rv_name)))
-        except AttributeError:
-            pass
-
-        return tree_node
+class AutoNameVisitor(ast.NodeVisitor):
+    def __init__(self):
+        self.random_variable_names = []
```

Comment:
Don't you have to call `super()`?

Reply:
In fact, it might be better to not overload `__init__` and just return `random_variable_names` from visit.

Reply:
So you mean, overload the `visit` function? How would we pass the names from the `visit_Assign` calls to the overall `visit` call?

Reply:
I've added a `super(...).__init__()` call for now.

Reply:
No, keep it like it is but return something in `visit_Assign`, if that's possible. That you wouldn't need the state and could delete `__init__`.

Reply:
Hm, I don't think that's how the `visit_Assign` method works: it gets called once per assignment statement (or in our case, once per `yield`), not once per model parsing, so the only thing it could do is return one name at a time, which isn't what we want. I think that overriding `__init__` is the closest thing we have.

Reply:
You're right.